### PR TITLE
Migrate BatchLabels to DX

### PR DESCRIPTION
**Pull request by toropok**
## Description of the issue/feature this PR addresses

This PR migrates the AT based Batch Labels to Dexterity

## Current behavior before PR

Batch Labels are based on the Archetype Framework

## Desired behavior after PR is merged

Batch Labels are based on the Dexterity Framework

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

**Comment by toropok** [comment]
@xispa hi pls checkup

File: src/senaite/core/profiles/default/types/BatchLabel.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  <!-- Title and Description -->
+  <property name="title" i18n:translate="">Batch Label</property>
+  <property name="description" i18n:translate=""/>
+
+  <!-- content-type icon -->
+  <property name="icon_expr">senaite_theme/icon/batchlabel</property>
+
+  <!-- factory name; usually the same as type name -->
+  <property name="factory">ContainerType</property>
```
Comment (xispa) [inline src/senaite/core/profiles/default/types/BatchLabel.xml]: typo. `ContainerType` --> `BatchLabel`

File: src/senaite/core/browser/controlpanel/batchlabels/view.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                    default="All"
+                ),
+                "contentFilter": {},
+                "columns": self.columns.keys(),
+            },
+        ]
+
+    def folderitem(self, obj, item, index):
+        item["replace"]["Title"] = get_link(
+            item["url"], item["Title"])
```
Comment (xispa) [inline src/senaite/core/browser/controlpanel/batchlabels/view.py]: Better use `get_link_for(obj)`

File: src/senaite/core/browser/controlpanel/batchlabels/view.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            default="Batch Labels")
+        )
+
+        self.icon = api.get_icon("BatchLabels", html_tag=False)
+        self.description = ""
+
+        self.columns = collections.OrderedDict((
+            ("Title", {
+                "title": _(
+                    "listing_batchlabels_column_title",
```
Comment (xispa) [inline src/senaite/core/browser/controlpanel/batchlabels/view.py]: Your linter will probably complain that is not py3-friendly, but better revert `i18n` messages to `unicode`. See https://github.com/senaite/senaite.core/pull/2566#discussion_r1621408893

**Review by xispa — CHANGES_REQUESTED** [review]
Thanks @toropok, I've left two comments

File: src/senaite/core/profiles/default/types/BatchLabel.xml (hunk 1 above)
Comment (toropok) [reply src/senaite/core/profiles/default/types/BatchLabel.xml]: damn, my mistake

File: src/senaite/core/browser/controlpanel/batchlabels/view.py (hunk 2 above)
Comment (toropok) [reply src/senaite/core/browser/controlpanel/batchlabels/view.py]: ok

File: src/senaite/core/browser/controlpanel/batchlabels/view.py (hunk 3 above)
Comment (toropok) [reply src/senaite/core/browser/controlpanel/batchlabels/view.py]: sure. Does it mean we should use 'u' prefix for any request to messagefactories everywhere?

**Review by xispa — APPROVED** [review]
Thanks!

**Pull request merged**